toolchain/check/type_structure.h

```diff
@@ -75,18 +84,35 @@ class TypeStructure : public Printable<TypeStructure> {
     Symbolic,
   };
 
+  struct ConcreteNoneType {
+    friend auto operator==(ConcreteNoneType /*a*/, ConcreteNoneType /*b*/)
+        -> bool {
+      return true;
+    }
```

Comment:
I think you can use `default`, i.e.:
```suggestion
    friend auto operator==(ConcreteNoneType lhs,
                           ConcreteNoneType rhs) -> bool = default;
```

Reply:
Done